### PR TITLE
placement: supports survival preferences

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/38605

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Documentation

- [x] Affects user behaviors
- [x] Contains syntax changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
placement: supports survival preferences
```

### CODE REVIEW

File: domain/infosync/placement_manager.go [hunk 1]
```diff
@@ -72,6 +74,7 @@ func (m *PDPlacementManager) PutRuleBundles(ctx context.Context, bundles []*plac
 		return err
 	}
 
+	log.Info("[Debug] put placement rules", zap.String("rules", string(b)))
```
Comment (xhebox) [inline domain/infosync/placement_manager.go:77]: Maybe `log.Debug`?

File: ddl/placement_policy_test.go [hunk 2]
```diff
@@ -588,12 +590,16 @@ func TestCreateTableWithPlacementPolicy(t *testing.T) {
...
+		"SURVIVAL_PREFERENCES=\"[region, zone]\"")
+	tk.MustExec("create placement policy z " +
+		"FOLLOWERS=1 " +
+		"SURVIVAL_PREFERENCES=\"[region, zone]\"")
 	tk.MustExec("create placement policy y " +
 		"FOLLOWERS=3 " +
 		"CONSTRAINTS=\"[+region=bj]\" ")
 	tk.MustExec("create table t(a int)" +
-		"PLACEMENT POLICY=\"x\"")
+		"PLACEMENT POLICY=\"z\"")
```
Comment (xhebox) [inline ddl/placement_policy_test.go:602]: Why not just add another `tk.mustExec()`.

File: ddl/placement/bundle.go [hunk 3]
```diff
@@ -124,6 +125,18 @@ func NewBundleFromConstraintsOptions(options *model.PlacementSettings) (*Bundle,
 		}
 	}
 
+	survivalPreferenceStr := options.SurvivalPreferences
+	if len(survivalPreferenceStr) > 0 {
+		labels := []string{}
+		err := yaml.UnmarshalStrict([]byte(survivalPreferenceStr), &labels)
+		if err != nil {
+			return nil, ErrInvalidConstraintsFormat
+		}
```
Comment (xhebox) [inline ddl/placement/bundle.go:134]: I would like to see an new error, or possibly a new function for parsing `survivalPreference`.

**Review by xhebox — COMMENTED** [review]
I think an update to https://github.com/pingcap/tidb/blob/master/docs/design/2020-06-24-placement-rules-in-sql.md is needed.

**Review by xhebox — APPROVED** [review]

File: docs/design/2020-06-24-placement-rules-in-sql.md [hunk 4]
```diff
@@ -413,6 +413,26 @@ If a table is imported when `tidb_placement_mode='IGNORE'`, and the placement po
 
 The default value for `tidb_placement_mode` is `STRICT`. The option is an enum, and in future we may add support for a `WARN` mode.
 
+
+#### survival preference
```
Comment (JmPotato) [inline docs/design/2020-06-24-placement-rules-in-sql.md:417]:
```suggestion
#### Survival preference
```

**Review by JmPotato — APPROVED** [review]

**Pull request merged**